### ISSUE TITLE
Doesn't work with LWJGL 3.3.3

### ISSUE THREAD

**Issue by theoparis** [xCollateral/VulkanMod#428]
```
java.lang.RuntimeException: Mixin transformation of net.minecraft.class_1041 failed
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:427)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218)
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:119)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
	at com.mojang.blaze3d.platform.GLX._initGlfw(GLX.java:66)
	at com.mojang.blaze3d.systems.RenderSystem.initBackendSystem(RenderSystem.java:848)
	at net.minecraft.class_310.<init>(class_310.java:512)
	at net.minecraft.client.main.Main.main(Main.java:223)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:100)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:129)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70)
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:422)
	... 14 more
Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: Critical injection failure: Redirector redirect2()Lorg/lwjgl/opengl/GLCapabilities; in vulkanmod.mixins.json:window.WindowMixin from mod vulkanmod failed injection check, (0/1) succeeded. Scanned 1 target(s). Using refmap VulkanMod_1.20.4-refmap.json
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.postInject(InjectionInfo.java:468)
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1384)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1062)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:402)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:327)
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:422)
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:403)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363)
	... 17 more

```

**Comment by Fox2Code** [comment]
I'm pretty sure it doesn't have anything to do with LWJGL version.

**Comment by xCollateral** [comment]
While this is not the full log, I suspect you are using the wrong mod version. Make sure mod's Minecraft version matches the version of Minecraft you are trying to run.

**Comment by KweezyCode** [comment]
it is surely broken

```
---- Minecraft Crash Report ----
// Uh... Did I do that?

Time: 2024-07-15 15:00:10
Description: Initializing game

java.lang.RuntimeException: Could not execute entrypoint stage 'client' due to errors, provided by 'vulkanmod'!
	at net.fabricmc.loader.impl.FabricLoaderImpl.lambda$invokeEntrypoints$2(FabricLoaderImpl.java:388)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:386)
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:53)
	at net.minecraft.class_310.<init>(class_310.java:477)
	at net.minecraft.client.main.Main.main(Main.java:239)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
Caused by: java.lang.NullPointerException
	at org.lwjgl.system.Checks.check(Checks.java:188)
	at org.lwjgl.glfw.GLFW.nglfwGetVideoMode(GLFW.java:1738)
	at org.lwjgl.glfw.GLFW.glfwGetVideoMode(GLFW.java:1761)
	at net.vulkanmod.config.video.VideoModeManager.getCurrentVideoMode(VideoModeManager.java:44)
	at net.vulkanmod.config.video.VideoModeManager.init(VideoModeManager.java:20)
	at net.vulkanmod.Initializer.onInitializeClient(Initializer.java:31)
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:384)
	... 6 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.fabricmc.loader.impl.FabricLoaderImpl.lambda$invokeEntrypoints$2(FabricLoaderImpl.java:388)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:386)
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:53)
	at net.minecraft.class_310.<init>(class_310.java:477)

-- Initialization --
Details:
	Modules: 
Stacktrace:
	at net.minecraft.client.main.Main.main(Main.java:239)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

-- System Details --
Details:
	Minecraft Version: 1.21
	Minecraft Version ID: 1.21
	Operating System: Linux (amd64) version 6.7.0-3-rt6-MANJARO
	Java Version: 21.0.3, Microsoft
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Microsoft
	Memory: 278871920 bytes (265 MiB) / 436207616 bytes (416 MiB) up to 436207616 bytes (416 MiB)
	CPUs: 4
	Processor Vendor: GenuineIntel
	Processor Name: Intel(R) Core(TM) i3-10110U CPU @ 2.10GHz
	Identifier: Intel64 Family 6 Model 142 Stepping 12
	Microarchitecture: Whiskey Lake
	Frequency (GHz): 2.10
	Number of physical packages: 1
	Number of physical CPUs: 2
	Number of logical CPUs: 4
	Graphics card #0 name: CometLake-U GT2 [UHD Graphics]
	Graphics card #0 vendor: Intel Corporation (0x8086)
	Graphics card #0 VRAM (MiB): 256.00
	Graphics card #0 deviceId: 0x9b41
	Graphics card #0 versionInfo: unknown
	Virtual memory max (MiB): 15879.11
	Virtual memory used (MiB): 5654.21
	Swap memory total (MiB): 12035.49
	Swap memory used (MiB): 1275.86
	Space in storage for jna.tmpdir (MiB): available: 81366.58, total: 243889.14
	Space in storage for org.lwjgl.system.SharedLibraryExtractPath (MiB): available: 81366.58, total: 243889.14
	Space in storage for io.netty.native.workdir (MiB): available: 81366.58, total: 243889.14
	Space in storage for java.io.tmpdir (MiB): available: 753.07, total: 768.72
	Space in storage for workdir (MiB): available: 81366.58, total: 243889.14
	JVM Flags: 12 total; -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M -XX:+DisableExplicitGC -XX:+AlwaysPreTouch -XX:+ParallelRefProcEnabled -Xms400M -Xmx400M -Xss1M
	Launched Version: Fabric 1.21
	Launcher name: java-minecraft-launcher
	Backend library: LWJGL version 3.3.3-snapshot
	Backend API: Unknown
	Window size: <not initialized>
	GFLW Platform: <error>
	GL Caps: Using framebuffer using OpenGL 3.2
	GL debug messages: <disabled>
	Is Modded: Definitely; Client brand changed to 'fabric'
	Universe: 404
	Type: Client (map_client.txt)
	Locale: ru_RU
	System encoding: UTF-8
	File encoding: UTF-8
	CPU: Intel(R) Core(TM) i3-10110U CPU @ 2.10GHz
```

**Comment by KweezyCode** [comment]
please reopen
